File: src/main/java/tk/hiddenname/smarthome/entity/hardware/AvailableGpioPins.java
package tk.hiddenname.smarthome.entity.hardware;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

import java.util.Set;

@Getter
@ToString
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class AvailableGpioPins {

    private final Set<Integer> availableGpioPins;
}

File: src/main/java/tk/hiddenname/smarthome/model/hardware/AvailableGpioPins.kt
package tk.hiddenname.smarthome.model.hardware

import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(SnakeCaseStrategy::class)
data class AvailableGpioPins(val availableGpioPins: Set<Int>)